br/pkg/restore/client.go

```diff
@@ -459,11 +509,33 @@ func (rc *Client) GoCreateTables(
 	}
 	outCh := make(chan CreatedTable, len(tables))
 	rater := logutil.TraceRateOver(logutil.MetricTableCreatedCounter)
+
+	var err error = nil
+
+	if rc.batchDllSize > 1 {
+
+		err = rc.createTablesInWorkerPool(ctx, dom, tables, dbPool, newTS, outCh)
+
+		if err == nil {
+			log.Info("bulk create tables success.")
+			defer close(outCh)
```

Comment:
this defer seems unnecessary?

Reply:
I'm afraid that the current implementation may cause creating tables and restoring sequential again. Maybe we could create a new goroutine for creating the tables like:

```go
go func() {
  defer close(outCh)
  if err := rc.createTablesInWorkerPool(ctx, dom, tables, dbPool, newTS, outCh); err != nil {
    errCh <- err
  }
}
```

(I'm not sure whether this is better than now, because creating table is fast enough for now, pipelining creating table and restoring isn't so meaningful comparing to before, and may make competition over resources and made unknown side effective.)